lib/nyxx.dart

```diff
@@ -151,6 +151,9 @@ export 'src/internal/shard/shard.dart' show IShard;
 export 'src/internal/shard/shard_manager.dart' show IShardManager;
 export 'src/typedefs.dart' show RawApiMap;
 export 'src/utils/enum.dart' show IEnum;
+export 'src/internal/http/http_route.dart' show HttpRoute;
+export 'src/internal/http/http_route_part.dart' show HttpRoutePart;
+export 'src/internal/http/http_route_param.dart' show HttpRouteParam;
```

Comment:
whats the point in exporting these classes? It seems like it should be only for internal use

Reply:
They are used to construct routes for making custom HTTP requests which is something nyxx should support - it also allows us to start using this new system in other nyxx packages.

Reply:
> it also allows us to start using this new system in other nyxx packages.

In other nyxx packages internal classes should be imported directly if needed. 

>They are used to construct routes for making custom HTTP requests which is something nyxx should support 

only way to make requests via nyxx should be using predifined requests like in IHttpEndpoints